File: C_Programming/LAB1ADV.cpp
#include <iostream>
using namespace std;

int lengthOfName; //defining variables
lengthOfName = name.length();
int value;
value=0;
int counter = 0;

int main() {
    string name;
    cout << "Please enter your first name: ";
    getline(cin, name); // take user input through cin doesnt like spaces so we wouldnt be abke to replace them with colons later on

    string ID;
    cout << "Please enter your full student ID: ";
    getline(cin, ID);
}



int check() {
for (int i=0; i<ID.length(); i++ ){ //Iterates over 
    char emptyArray[i];
    emptyArray[0]=ID[i];
    int emptyList = atoi(emptyArray)

    if (emptyList[i] >2; and emptyList[i] < 7) { //checks to see if value is between 2 and 7
        value = emptylist[i]; //aassigns the inetger to our variable
        break;
    }
if (value == 0){ // if there is no integer in student ID with this condition it assumes it is 4
    value = 4;
}
}

for (int  i=0; i<lengthOfName; i++){
    if (name[i]==" "){ //if a character in the name is a space then
        name[i]=';'; //repalce it with a colon
    }
    else{
        if (name[i].isupper()){ // checks to see if the character is uppercase
            name[i]=name[i].lower(); // if it is then make it lower case
            cout << name[i]; << endl; // output
        }

        else{
            name[i]=name[i].upper();//if its not upper then it must be lowercase
            cout << name[i] << endl; // so make it upper case and output
        }
    }
}
}

if (count == value){
    cout << endl;
    count=0;//reset counter
}
if (count < value){ //if counter is less than value then padd it with "="
    int n = value - (count % value);
    for (int i=0; i<n; i++){
        cout << '=';
    }
    cout << endl;
}

count = 0; // reset counter

File: Boring/Python/LAB1ADV.py
name = input("Enter your name: ") #Takes user input for their name
ID = input("Enter your student ID: ") #declares a variable called ID
LengthName=len(name)#fidns length of user input name and declares a new variable
value = 0 #declares a variable with the value of 0
for i in ID:
    if int(i) > 2 and int(i) < 7: #checks to see if an integer in ID is between 2 and 7
        value = int(i) #if it is between 2 and 7 it assigns it to value
        break
if value == 0: #if there are no integers in ID between 2 and 7 it assumes it is 4
    value = 4


for i in range(1, len(name)+1):
    if name[i-1]==" ": #if there is a space character in the name then
        print(":", end="")#replace it with a smei colon
    else:
        if name[i-1].isupper():#checks if a character in name is an uppercase
            print(name[i-1].lower(), end="")#if it is turn it to a lower case

        else:
            print(name[i-1].upper(), end="")# if its a lower case then turn it to an upper case

    if i % value == 0:
        print("\n")#prints newline if the remainer of i mod value is 0


if (LengthName % value):
    padding = value - (LengthName % value)
    print("="*padding, end="") #if the name cant fill the value size then fill the gaps with a "=" symbol
